backend/index.ts

```diff
@@ -1,74 +1,157 @@
-    res.status(500).send('Failed to get memes :(');
+      const result = await poolConnection
+        .request()
+        .input("ChannelName", sql.NVarChar, channel)
+        .input("Count", sql.Int, count).query(`
+          SELECT TOP (@Count) Id, Name, Author, Username, AuthorImage, Date, Url, Type, Reactions, ChannelName
+          FROM MediaFiles
+          WHERE ChannelName = @ChannelName
+          ORDER BY Date DESC
+        `);
+
+      const parsedRecords = result.recordset.map((record) =>
+        toCamelCaseKeys(record)
+      );
+
+      res.json(parsedRecords);
+    } catch (error) {
+      console.error("Failed to get memes:", error);
+      res.status(500).send("Failed to get memes :(");
+    }
   }
-});
+);
+
+app.get(
+  "/latest-blasts",
+  authenticate,
+  [
+    query("count")
+      .optional()
+      .isInt({ min: 1, max: 10 })
+      .withMessage("Count should be an integer between 1 and 10"),
+  ],
+  async (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const count = parseInt(req.query.count as string, 10) || 5;
+
+    try {
+      const poolConnection = await poolPromise;
 
-app.get('/latest-blasts', authenticate, async (req: Request, res: Response) => {
-  const count = parseInt(req.query.count as string, 10) || 5;
+      const channelNames = ["korktavla", "online"];
 
-  if (count > 10 || count < 1) return res.status(400).send('Count should be between 1 and 10');
+      const result = await poolConnection
+        .request()
+        .input("Count", sql.Int, count)
+        .input("Channel1", sql.NVarChar, channelNames[0])
+        .input("Channel2", sql.NVarChar, channelNames[1]).query(`
+          SELECT TOP (@Count) Id, Text, Author, AuthorImage, Date, ChannelName
+          FROM MediaFiles
+          WHERE ChannelName IN (@Channel1, @Channel2)
+          ORDER BY Date DESC
+        `);
 
-  try {
-    await manageDirectory(textDir);
-    await fetchTextMessagesFromChannels(channels.blasts, count, req);
-    const textMetadata = await listFiles(textDir);
-    res.json(textMetadata);
-  } catch (error) {
-    console.error('Failed to get blasts:', error);
-    res.status(500).send('Failed to get blasts :(');
+      const parsedRecords = result.recordset.map((record) =>
+        toCamelCaseKeys(record)
+      );
+
+      res.json(parsedRecords);
+    } catch (error) {
+      console.error("Failed to get blasts:", error);
+      res.status(500).send("Failed to get blasts :(");
+    }
   }
-});
+);
+
+app.get(
+  "/movember",
+  authenticate,
+  [
+    query("date")
+      .optional()
+      .isISO8601()
+      .withMessage("Date must be in ISO8601 format (YYYY-MM-DD)"),
+  ],
+  async (req: Request, res: Response) => {
+    try {
+      const poolConnection = await poolPromise;
+      const channel = "movember";
+      const cutoffDate = req.query.date || "2024-11-30";
```

Comment:
Burde vel ikke hardkode 2024

Reply:
Var mest for å ikke gjøre så mye mer endringer i frontenden. Men kan oppdatere frontenden også,